### PR TITLE
adding resources in org eventbridge module

### PR DESCRIPTION
**Pull request by sameer-in**
- adding resources in org eventbridge module

### CODE REVIEW

File: modules/services/trust-relationship/outputs.tf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 output "cspm_role_arn" {
-  value       = aws_iam_role.cspm_role[0].arn
+  value       = aws_iam_role.cspm_role.arn
```
Comment (nkraemer-sysdig) [inline modules/services/trust-relationship/outputs.tf:2]: I don't think we need an output anymore

File: modules/services/event-bridge/organizational.tf [hunk 2]
```diff
@@ -8,14 +8,91 @@ data "aws_organizations_organization" "org" {
...
 
+data "aws_region" "current" {}
+
 locals {
   organizational_unit_ids = var.is_organizational && length(var.organization_units) == 0 ? [for root in data.aws_organizations_organization.org[0].roots : root.id] : toset(var.organization_units)
 }
 
+# permission policy for stackset admin role
+resource "aws_iam_policy" "admin_role_policy" {
+  name   = "stackset-admin-role-policy"
```
Comment (nkraemer-sysdig) [inline modules/services/event-bridge/organizational.tf:19]: Do we need `count = var.is_organizational ? 1 : 0` here?

File: modules/services/event-bridge/organizational.tf [hunk 3]
```diff
@@ -8,14 +8,91 @@ data "aws_organizations_organization" "org" {
...
+            "Effect": "Allow"
+        }
+    ]
+}
+EOF
+}
+
+# policy attachment for stackset admin role
+resource "aws_iam_role_policy_attachment" "admin_role_attachment" {
+  policy_arn = aws_iam_policy.admin_role_policy.arn
```
Comment (nkraemer-sysdig) [inline modules/services/event-bridge/organizational.tf:40]: Do we need a `count = var.is_organizational ? 1 : 0` here?

**Review by nkraemer-sysdig — COMMENTED** [review]
Seems good for the CSPM only and Full Install flows, but I think the EventBridge only flow is missing a trustRelationship in the management account that is needed to scrape the org structure. Not sure how to include this though, since adding it to the eventBridge module will cause it to conflict with the trustRelationship module in a full install. 

Do you have an example of a snippet for all three installs? (CSPM only, EB only and Full Install)

File: modules/services/trust-relationship/outputs.tf (hunk 1 above)
Comment (sameer-in) [reply modules/services/trust-relationship/outputs.tf]: removed the output as its not needed

File: modules/services/event-bridge/organizational.tf (hunk 3 above)
Comment (sameer-in) [reply modules/services/event-bridge/organizational.tf]: added count here.

File: modules/services/event-bridge/organizational.tf (hunk 2 above)
Comment (sameer-in) [reply modules/services/event-bridge/organizational.tf:21]: done. added a count.

**Comment by sameer-in** [comment]
@nkraemer-sysdig As we need a role, which is exactly like CSPMRole, we can add it as a separate module for EBOrg. 
```
provider "aws" {
  region = "us-west-2"
}

module "cspm-role" {
  source           = "draios/secure-for-cloud/aws//modules/services/trust-relationship"
  role_name        = "sameer-cspm-role1"
  trusted_identity = "arn:aws:iam::059797578166:role/cloudbench-trusted-role"
  external_id      = "c4ca4238a0b923820dcc509a6f75849b"
}

module "secure-for-cloud_example_org-event-bridge" {
  source           = "../../terraform-aws-secure-for-cloud/modules/services/event-bridge"
  target_event_bus_arn = "arn:aws:events:us-east-1:064689838359:event-bus/us-east-1-integration01-falco-1"
  is_organizational = true
  trusted_identity = "arn:aws:iam::064689838359:role/us-east-1-integration01-secure-assume-role"
  external_id      = "c4ca4238a0b923820dcc509a6f75849b"
  organization_units = ["r-op65"]
  regions = ["us-west-1"]

}
```

**Comment by haresh-suresh** [comment]
@sameer-in Do you want to add the CSPM role module in this PR for EB ? 
Do we have any other way to handle the duplicate role provisioning problem ? Could we have some other conditionals to handle that case ?

**Comment by sameer-in** [comment]
@haresh-suresh I think we don't need to add role in EB. as we already have a separate module for it (cspm). Which role duplication scenario are we discussing?

**Comment by haresh-suresh** [comment]
> @haresh-suresh I think we don't need to add role in EB. as we already have a separate module for it (cspm). Which role duplication scenario are we discussing?

We discussed offline, @sameer-in do you not need to add the commit for full install in this PR as well ?

File: modules/services/trust-relationship/main.tf [hunk 4]
```diff
@@ -39,18 +39,14 @@ data "aws_iam_policy_document" "trust_relationship" {
...
   assume_role_policy = data.aws_iam_policy_document.trust_relationship.json
   tags               = var.tags
 }
 
 
 resource "aws_iam_role_policy_attachment" "cspm_security_audit" {
-  count = var.is_organizational && !var.provision_caller_account ? 0 : 1
-
-  role       = aws_iam_role.cspm_role[0].id
+  role       = aws_iam_role.cspm_role.id
```
Comment (haresh-suresh) [inline modules/services/trust-relationship/main.tf:49]: To be safe / consistent, let's make all role creation to be inline and have no separate policy attachment resource.
What do you think ?

Comment (sameer-in) [reply modules/services/trust-relationship/main.tf:49]: I don't think its an issue for single install.

File: modules/services/event-bridge/main.tf [hunk 5]
```diff
@@ -50,9 +49,10 @@ EOF
 # Target to forward all CloudTrail events to Sysdig's EventBridge Bus.
 # See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_event_target#cross-account-event-bus-target
 resource "aws_cloudwatch_event_target" "sysdig" {
-  count      = var.is_organizational && !var.provision_management_account ? 0 : 1
+  count      = var.is_organizational ? 0 : 1
   depends_on = [aws_iam_role.event_bus_invoke_remote_event_bus]
 
+  //    rule     = aws_cloudwatch_event_rule.sysdig.name
```
Comment (nkraemer-sysdig) [inline modules/services/event-bridge/main.tf:55]: Can we remove this line?

**Review by nkraemer-sysdig — DISMISSED** [review]

File: modules/services/event-bridge/main.tf (hunk 5 above)
Comment (sameer-in) [reply modules/services/event-bridge/main.tf]: done. removed this.

**Review by nkraemer-sysdig — APPROVED** [review]

File: modules/services/event-bridge/main.tf [hunk 6]
```diff
@@ -67,9 +66,9 @@ resource "aws_cloudwatch_event_target" "sysdig" {
 # Role that will be used by EventBridge when sending events to Sysdig's EventBridge Bus. The EventBridge service is
 # given permission to assume this role.
 resource "aws_iam_role" "event_bus_invoke_remote_event_bus" {
-  count = var.is_organizational ? var.provision_management_account ? 0 : 1 : var.deploy_global_resources ? 1 : 0
+  count = (var.is_organizational || var.deploy_global_resources) ? 1 : 0
 
-  name = var.name
+  name = join("-", [var.name, "MgmtEBRole"])
```
[the remark this answers is not in the document]
Comment (haresh-suresh) [reply modules/services/event-bridge/main.tf:71]: Just readability, when we were testing this, it was very helpful to quickly glance and distinguish between the stackset,  CSPM and EB roles.

**Pull request merged**